src/wallet/rpcwallet.cpp

```diff
@@ -3934,6 +3935,84 @@ UniValue autocombinerewards(const JSONRPCRequest& request)
+            "\nThis will set the auto-combine threshold value.\n"
+            "\nWallet will automatically monitor for any coins with value below the threshold amount, and combine them if they reside with the same PIVX address\n"
+            "When auto-combine runs it will create a transaction, and therefore will be subject to transaction fees.\n"
+
+            "\nArguments:\n"
+            "1. enable          (boolean, required) Enable auto combine (true) or disable (false)\n"
+            "2. threshold       (numeric, optional) Threshold amount (default: 0)\n"
+
+            "\nResult:\n"
+            "{\n"
+            "  \"enabled\": true|false,      (boolean) true if auto-combine is enabled, otherwise false\n"
+            "  \"threshold\": n.nnn,         (numeric) auto-combine threshold in PIV\n"
+            "  \"saved\": true|false         (boolean) true if setting was saved to the database, otherwise false\n"
+            "}\n"
+
+            "\nExamples:\n" +
+            HelpExampleCli("setautocombinethreshold", "500.12") + HelpExampleRpc("setautocombinethreshold", "500.12"));
+
+    EnsureWallet();
+
+    RPCTypeCheck(request.params, {UniValue::VBOOL, UniValue::VNUM});
+
+    bool fEnable = false;
+    if (!request.params.empty())
+        fEnable = request.params[0].get_bool();
+
+    CWalletDB walletdb(pwalletMain->GetDBHandle());
+    CAmount nThreshold = 0;
+
+    if (fEnable && request.params.size() > 1)
+        nThreshold = AmountFromValue(request.params[1]);
+
+    {
+        LOCK(pwalletMain->cs_wallet);
+        pwalletMain->fCombineDust = fEnable;
+        pwalletMain->nAutoCombineThreshold = nThreshold;
+
+        UniValue result(UniValue::VOBJ);
+        result.pushKV("enabled", fEnable);
+        result.pushKV("threshold", ValueFromAmount(pwalletMain->nAutoCombineThreshold));
+        if (walletdb.WriteAutoCombineSettings(fEnable, nThreshold)) {
+            result.pushKV("saved", "true");
+        } else {
+            result.pushKV("saved", "false");
+        }
+
+        return result;
+    }
+}
+
+UniValue getautocombinethreshold(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 0)
+        throw std::runtime_error(
+            "getautocombinethreshold\n"
+            "\nReturns the current threshold for auto combining UTXOs, if any\n"
+
+            "\nResult:\n"
+            "{\n"
+            "  \"enabled\": true|false,        (boolean) true if auto-combine is enabled, otherwise false\n"
+            "  \"threshold\": n.nnn            (numeric) the auto-combine threshold amount in PIV\n"
+            "}\n"
+
+            "\nExamples:\n" +
+            HelpExampleCli("getautocombinethreshold", "") + HelpExampleRpc("getautocombinethreshold", ""));
+
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("enabled", pwalletMain->fCombineDust);
+    result.pushKV("threshold", ValueFromAmount(pwalletMain->nAutoCombineThreshold));
```

Comment:
we should lock `cs_wallet` before reading `fCombineDust` and `nAutoCombineThreshold`